types/field_type.go

```diff
@@ -231,8 +231,12 @@ func (ft *FieldType) CompactStr() string {
 			suffix = fmt.Sprintf("(%d", displayFlen)
 			if isDecimalNotDefault {
 				suffix += fmt.Sprintf(",%d", displayDecimal)
+			} else {
+				suffix += fmt.Sprintf(",%d", 0)
 			}
 			suffix += ")"
+		} else {
+			suffix = "(10,0)"
```

Comment:
the default fsp for tidb seem be not `10` which is different to mysql

https://github.com/lysu/tidb/commit/ea29a1c30dc89aa1874a9bcd1e796412cf6c7253#diff-af881ce5457dfb227fd7fb66090a339cR32